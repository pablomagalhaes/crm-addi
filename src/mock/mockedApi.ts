import { Model, Server, Response } from 'miragejs';

import { registries } from './data/registries.mock';
import { records } from './data/records.mock';
import { leads } from './data/leads.mock';
import { prospectors } from './data/prospectors.mock';

const mockedApi = () => {
  const api = new Server({
    namespace: 'api',

    models: {
      registry: Model,
      record: Model,
      lead: Model,
      prospector: Model,
    },

    seeds(server: any) {
      registries.forEach(registry => server.create('registry', registry));
      records.forEach(record => server.create('record', record));
      leads.forEach(lead => server.create('lead', lead));
      prospectors.forEach(prospector =>
        server.create('prospector', prospector)
      );
    },

    routes() {
      this.timing = 100;

      this.get(
        '/registries/:nationalIdNumber',
        (schema: any, request: any) => {
          const registries = schema.registries.where({
            nationalIdNumber: request.params.nationalIdNumber,
          });

          if (registries.length) return registries;

          return new Response(
            404,
            {},
            { status: 404, error: 'Registry not found' }
          );
        },
        { timing: 1000 }
      );

      this.get(
        '/records/:nationalIdNumber',
        (schema: any, request: any) => {
          const records = schema.records.where({
            nationalIdNumber: request.params.nationalIdNumber,
          });

          if (records.length) return records;

          return new Response(
            404,
            {},
            { status: 404, error: 'Registry not found' }
          );
        },
        { timing: 2500 }
      );

      this.get('/leads', (schema: any) => schema.leads.all());

      this.delete('/leads/:leadId', (schema: any, request) =>
        schema.leads.find(request.params.leadId).destroy()
      );

      this.get('/prospectors', (schema: any) => schema.prospectors.all());

      this.post('/prospectors', (schema: any, request: any) => {
        const attrs = JSON.parse(request.requestBody);
        delete attrs.id;

        return schema.prospectors.create(attrs);
      });

      this.post(
        '/score/:nationalIdNumber',
        (_, request: any) => {
          let score = Math.floor(Math.random() * 101);

          const {
            existsInRegisty,
            matchWithRegisty,
            hasJudicialRecord,
          } = JSON.parse(request.requestBody);

          if (existsInRegisty) score = score + 5;
          if (matchWithRegisty) score = score + 5;
          if (!hasJudicialRecord) score = score + 5;

          if (score >= 100) score = 100;

          return {
            nationalIdNumber: request.params.nationalIdNumber,
            score,
          };
        },
        { timing: 3000 }
      );
    },
  });

  return api;
};

export default mockedApi;
