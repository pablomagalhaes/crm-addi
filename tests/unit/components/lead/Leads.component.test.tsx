import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Leads from '@/components/lead/Leads.component';
import { leads as leadsDto } from '@/mock/data/leads.mock';
import { LeadModel } from '@/models/Lead.model';

const leads: LeadModel[] = leadsDto.map((leadDto, index) => ({
  id: `${index}`,
  ...leadDto,
}));

describe('<Leads />', () => {
  it('should match snapchat', () => {
    const wrapper = shallow(<Leads leads={leads} />);
    expect(wrapper).toMatchSnapshot();
  });
});
