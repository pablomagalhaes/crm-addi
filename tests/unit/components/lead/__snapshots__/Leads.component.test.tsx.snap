// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Leads /> should match snapchat 1`] = `
<styled.table>
  <thead>
    <tr>
      <th
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        Code
      </th>
      <th>
        Full Name
      </th>
      <th>
        ID Number
      </th>
      <th>
        Birthdate
      </th>
      <th>
        E-mail
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        National Registry
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Judicial
      </th>
      <th
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Credit Score
      </th>
      <th>
        Validation
      </th>
    </tr>
  </thead>
  <tbody>
    <Lead
      key="lead-43240154064"
      lead={
        Object {
          "birthdate": "1990-08-29",
          "email": "johndoe@gmail.com",
          "existsInRegisty": null,
          "firstName": "John",
          "hasJudicialRecord": null,
          "id": "0",
          "lastName": "Doe",
          "matchWithRegisty": null,
          "nationalIdNumber": "43240154064",
          "score": 0,
        }
      }
    />
    <Lead
      key="lead-74186516006"
      lead={
        Object {
          "birthdate": "1988-05-21",
          "email": "mariaxavier@gmail.com",
          "existsInRegisty": null,
          "firstName": "Maria",
          "hasJudicialRecord": null,
          "id": "1",
          "lastName": "Xavier",
          "matchWithRegisty": null,
          "nationalIdNumber": "74186516006",
          "score": 0,
        }
      }
    />
    <Lead
      key="lead-83256381090"
      lead={
        Object {
          "birthdate": "1972-01-15",
          "email": "johnwick@gmail.com",
          "existsInRegisty": null,
          "firstName": "John",
          "hasJudicialRecord": null,
          "id": "2",
          "lastName": "Wick",
          "matchWithRegisty": null,
          "nationalIdNumber": "83256381090",
          "score": 0,
        }
      }
    />
    <Lead
      key="lead-19399063003"
      lead={
        Object {
          "birthdate": "1972-01-15",
          "email": "danielsmith@gmail.com",
          "existsInRegisty": null,
          "firstName": "Daniela",
          "hasJudicialRecord": null,
          "id": "3",
          "lastName": "Smith",
          "matchWithRegisty": null,
          "nationalIdNumber": "19399063003",
          "score": 0,
        }
      }
    />
    <Lead
      key="lead-70143939017"
      lead={
        Object {
          "birthdate": "1990-07-03",
          "email": "ashlynreeve@gmail.com",
          "existsInRegisty": null,
          "firstName": "Ashlyn",
          "hasJudicialRecord": null,
          "id": "4",
          "lastName": "Reeve",
          "matchWithRegisty": null,
          "nationalIdNumber": "70143939017",
          "score": 0,
        }
      }
    />
    <Lead
      key="lead-89896287090"
      lead={
        Object {
          "birthdate": "1977-12-14",
          "email": "valentinolegge@gmail.com",
          "existsInRegisty": null,
          "firstName": "Valentino",
          "hasJudicialRecord": null,
          "id": "5",
          "lastName": "Legge",
          "matchWithRegisty": null,
          "nationalIdNumber": "89896287090",
          "score": 0,
        }
      }
    />
  </tbody>
</styled.table>
`;
